examples/AspNetCore/Program.cs

```diff
@@ -24,27 +24,28 @@
 
 var builder = WebApplication.CreateBuilder(args);
 
-var serviceName = "AspNetCoreExampleService";
-
 // OpenTelemetry
 var assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "unknown";
 
 // Switch between Zipkin/Jaeger/OTLP by setting UseExporter in appsettings.json.
 var tracingExporter = builder.Configuration.GetValue<string>("UseTracingExporter").ToLowerInvariant();
 
-var resourceBuilder = tracingExporter switch
+var serviceName = tracingExporter switch
 {
-    "jaeger" => ResourceBuilder.CreateDefault().AddService(builder.Configuration.GetValue<string>("Jaeger:ServiceName"), serviceVersion: assemblyVersion, serviceInstanceId: Environment.MachineName),
-    "zipkin" => ResourceBuilder.CreateDefault().AddService(builder.Configuration.GetValue<string>("Zipkin:ServiceName"), serviceVersion: assemblyVersion, serviceInstanceId: Environment.MachineName),
-    "otlp" => ResourceBuilder.CreateDefault().AddService(builder.Configuration.GetValue<string>("Otlp:ServiceName"), serviceVersion: assemblyVersion, serviceInstanceId: Environment.MachineName),
-    _ => ResourceBuilder.CreateDefault().AddService(serviceName, serviceVersion: assemblyVersion, serviceInstanceId: Environment.MachineName),
+    "jaeger" => builder.Configuration.GetValue<string>("Jaeger:ServiceName"),
+    "zipkin" => builder.Configuration.GetValue<string>("Zipkin:ServiceName"),
+    "otlp" => builder.Configuration.GetValue<string>("Otlp:ServiceName"),
+    _ => "AspNetCoreExampleService",
 };
 
+Action<ResourceBuilder> configureResources = r => r.AddService(
+    serviceName, serviceVersion: assemblyVersion, serviceInstanceId: Environment.MachineName);
+
 // Traces
 builder.Services.AddOpenTelemetryTracing(options =>
 {
     options
-        .SetResourceBuilder(resourceBuilder)
+        .ConfigureResources(configureResources)
```

Comment:
Make the name singular `ConfigureResource`?

I think this better matches things conceptually in that each tracer/meter/logging provider has a single resource associated with it once it's built. I think we create multiple `Resource` objects as the resource is being built, but then they all get merged into one.

Reply:
No strong opinion, but the plural was not an accident. Aren't the semantic conventions for resources each called a resoruce? E.g. you add the Telemetry SDK resource, you add the service resource and the process.runtime resource. Of course in the end it's merged in a single resource.

Also, the primary feature of this new method is that it can be called multiple times additively. I felt that this is a bit better conveyed by the plural.

Singular also makes sense though, and I'm happy to change it if you think that's better.

Reply:
I think if the idea is to be able to call it multiple times to configure multiple resources, having it singular seems more natural to me. Also if it's plural, I would assume to pass a list of things so another point for singular.

Reply:
> Aren't the semantic conventions for resources each called a resoruce? E.g. you add the Telemetry SDK resource, you add the service resource and the process.runtime resource.

Hmm, interesting. It hasn't occurred to me to think of it this way. Maybe there's not a "correct" interpretation here, but it's language like this from the [Resource SDK spec](https://github.com/open-telemetry/opentelemetry-specification/blob/9abbdd39d0b35f635f833f072013431da419894e/specification/resource/sdk.md) that has led me to my interpretation

> A [Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#resources) is an immutable representation of the entity producing telemetry as [Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute). For example, a process producing telemetry that is running in a container on Kubernetes has a Pod name, it is in a namespace and possibly is part of a Deployment which also has a name. All three of these attributes can be included in the Resource.

That's to say a single resource is defined and is comprised of attributes from potentially multiple bodies of semantic conventions.